README.md

```diff
@@ -1,19 +1,127 @@
+
+- Fix https://github.com/tc39/ecma402/issues/238 (currency long name has dependency on plural form, and the currency long name pattern has dependency on currencyWidth).
+- Move pattern resolution out of the constructor to keep all internal fields of NumberFormat locale-agnostic, making it easier to reason about behavior in the format method.
+
+In addition, one missing option is added to the existing `currencyDisplay` setting: "narrowSymbol", which uses the CLDR narrow-format symbol.
+
+## II. Measure Units
+
+Units of measurement can be formatted as follows:
+
+```javascript
+(9.81).toLocaleString("en-US", {
+    style: "unit",
+    unit: "acceleration-meter-per-second-squared",
+    unitDisplay: "short"
+});
+// ==> "9.81 m/s²"
+```
+
+The syntax was discussed in #3.
+
+- `style` receives the string value "unit"
+- `unit` receives a string measure unit identifier, defined in [UTS #35](http://unicode.org/reports/tr35/tr35-general.html#Unit_Elements).  See also the [full list of unit identifiers](https://unicode.org/repos/cldr/tags/latest/common/validity/unit.xml).
+- `unitDisplay`, named after the corresponding setting for currencies, `currencyDisplay`, takes either "narrow", "short", or "long".
+
+## III. Scientific and Compact Notation
+
+Scientific and compact notation are represented by the new option `notation` and can be formatted as follows:
+
+```javascript
+(987654321).toLocaleString("en-US", {
+    notation: "scientific"
+});
+// ==> 9.877E8
+
+(987654321).toLocaleString("en-US", {
+    notation: "engineering"
+});
+// ==> 987.7E6
+
+(987654321).toLocaleString("en-US", {
+    notation: "compact",
+    compactDisplay: "long"
+});
+// ==> 987.7 million
+```
+
+The syntax was discussed in #5.
+
+- `notation` takes either "scientific", "engineering", "compact", or "plain"
+- `compactDisplay`, used only when `notation` is "compact", takes either "short" or "long"
+
+Rounding-related settings (min/max integer/fraction digits) are applied after the number is scaled according to the chosen notation.
+
+Notation styles are allowed to be combined with other options.  (ICU supports this.)
+
+```javascript
+(299792458).toLocaleString("en-US", {
+    notation: "scientific",
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+    style: "unit",
+    unit: "speed-meter-per-second"
+});
+// ==> 3.00E8 m/s
+```
+
+## IV. Sign Display
+
+The sign can be displayed on positive numbers:
+
+```javascript
+(55).toLocaleString("en-US", {
+    signDisplay: "always"
+});
+// ==> +55
+```
+
+Currency accounting sign display is also supported via a new option:
+
+```javascript
+(-55).toLocaleString("en-US", {
+    style: "currency",
+    currency: "USD",
+    currencySignDisplay: "accounting"
+});
+// ==> ($55.00)
+```
+
+The syntax was discussed in #6.
+
+- `signDisplay` takes either "auto", "always", "never", or "except-zero"
+- `currencySignDisplay` takes either "standard" or "accounting"
```

Comment:
Can you explain here what these various options values mean, and what the default behavior is when they are missing? I don't think everyone reading this will know that "accounting" stands for using the parentheses, and "standard" stands for a - sign. Examples in non-en-US locales would also be nice.

Reply:
I added a table for the signDisplay options.

I fleshed out the paragraph about accounting format a bit more and changed it to a non-en-US locale.